source/vibe/db/postgresql/package.d

```diff
@@ -39,25 +41,37 @@ shared class PostgresClient
             afterStartConnectOrReset
         );
 
-        pool = new shared ConnectionPool!__Conn({ return new __Conn(settings); }, connNum);
+        pool = new ConnectionPool!__Conn({ return new __Conn(settings); }, connNum);
     }
 
     LockedConnection!__Conn lockConnection()
     {
-        import vibe.core.core: yield;
-
-        logDebugV("get connection from a shared pool");
-
-        LockedConnection!__Conn conn;
-
-        while(!pool.tryLockConnection(conn))
-        {
-            yield(); // TODO: Need to replace yield() by events handling in the conn pool for compatibility with threads and fibers both
-            continue;
-        }
+        logDebugV("get connection from the pool");
+        return new LockedConnection!__Conn(pool.lockConnection());
```

Comment:
What will happen If here is no free connections in the pool?

Reply:
fiber will block on the userspace semaphore.

Reply:
https://github.com/vibe-d/vibe-core/blob/master/source/vibe/core/connectionpool.d#L76

vibe-d:core should be allright, vibe-core had a bug in semaphore until https://github.com/vibe-d/vibe-core/pull/31

Reply:
Memo: do not forgot to increase vibe version